### PR TITLE
Snapshot Resource

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- Create, Import, Modify and Delete PowerMax Snapshots

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Snapshot Resources

##### ADDITIONAL INFORMATION
```
$ TF_LOG=INFO TF_ACC=1 go test -count=1 -v -run TestAccSnapshotResource
=== RUN   TestAccSnapshotResource
    provider_test.go:90:
                        provider "powermax" {
                                username      = "smc"
                                password      = "smc"
                                endpoint      = "http://localhost:3000"
                                serial_number = "000120000606"
                                pmax_version  = "100"
                                insecure      = true
                        }
--- PASS: TestAccSnapshotResource (39.01s)
PASS
ok      terraform-provider-powermax/powermax/provider   39.968s

```
## Make Check
```
$ make check
terraform fmt -recursive examples/
gofmt -s -w .
golangci-lint run --fix --timeout 5m
go vet

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: examples/resources/powermax_snapshot/provider.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#   endpoint      = var.endpoint
+#   serial_number = var.serial_number
+#   pmax_version  = var.pmax_version
+#   insecure      = true
+# }
+
+provider "powermax" {
+  username      = "smc"
+  password      = "smc"
+  endpoint      = "https://10.225.104.37:8443"
```
Comment (shekhar-j) [inline examples/resources/powermax_snapshot/provider.tf:21]: can you remove actual data?

**Review by shekhar-j — APPROVED** [review]

File: examples/resources/powermax_snapshot/provider.tf (hunk 1 above)
Comment (doriac11) [reply examples/resources/powermax_snapshot/provider.tf:21]: Yes sorry thought I cleaned this up after testing, will do!

File: examples/resources/powermax_snapshot/variables.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# variable "username" {
```
Comment (nikitajoshi1) [inline examples/resources/powermax_snapshot/variables.tf:1]: @doriac11 is this commented out for some reason?

**Review by nikitajoshi1 — APPROVED** [review]

File: examples/resources/powermax_snapshot/variables.tf (hunk 2 above)
Comment (doriac11) [reply examples/resources/powermax_snapshot/variables.tf:1]: Haha yea Just did while I was testing, I will uncomment. 
Thanks!

**Pull request merged**